jwst/stpipe/subproc.py

```diff
@@ -61,7 +61,7 @@ def process(self, *args):
         newargs = []
         for i, arg in enumerate(args):
             if isinstance(arg, datamodels.DataModel):
-                filename = "{0}.{1:04d}.{1}".format(
+                filename = "{0}.{1:04d}.{2}".format(
```

Comment:
And here is the good reason for harshness. Wow, this would have never worked.  *oof*

Reply:
Agreed.  I'm happy to accept harsh checking if it uncovers actual bugs, as is the case here.